### PR TITLE
Fix Issue 11714 - Improve error message for wrongly initialized thread-local class and pointer to struct instances

### PR DESCRIPTION
**Pull request by ibuclaw**
My attempt to take a stab at #2943

### CODE REVIEW

**Review by JinShil — APPROVED** [review]

File: test/fail_compilation/fail11714.d [hunk 1]
```diff
@@ -0,0 +1,21 @@
+/*
+TEST_OUTPUT:
+---
+fail_compilation/fail11714.d(14): Error: variable fail11714.c is a thread-local class and cannot have a static initializer. Use `static this()` to initialize instead.
+fail_compilation/fail11714.d(21): Error: variable fail11714.s is a thread-local pointer to struct and cannot have a static initializer. Use `static this()` to initialize instead.
```
Comment (JinShil) [inline test/fail_compilation/fail11714.d]: Would be nice if `fail11714.c` and `fail11714.s` had tick marks around them.

Comment (ibuclaw) [reply test/fail_compilation/fail11714.d]: That would be a change for another pr. 😉

Comment (ibuclaw) [reply test/fail_compilation/fail11714.d]: Also, every source file in the testsuite would need updating if you make that change.

**Comment by ibuclaw** [comment]
@WebDrake, mind having a quick look at this?

**Comment by WebDrake** [comment]
@ibuclaw looks good from my side.  It's a clear diagnosis of what's wrong and a clear indicator of what needs to be done to fix it.  Seems a straightforward improvement.

**Review by wilzbach — APPROVED** [review]

**Pull request merged**